lib/pengine/complex.c

```diff
@@ -596,16 +596,16 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
 
     if (id == NULL) {
         pe_err("Must specify id tag in <resource>");
-        return FALSE;
+        return pcmk_rc_error;
```

Comment:
We should avoid pcmk_rc_error except in dire circumstances (generally when we don't know what happened). For these, pcmk_rc_schema_validation or pcmk_rc_cib_corrupt would be acceptable. Since the callers only check vs pcmk_rc_ok it's not terribly important, but we want to give something reasonable if someone ever calls pcmk_rc_str() on the result.

In general, if we know the cause of the error, adding a new pcmk_rc_ code is preferable to using pcmk_rc_error.

Reply:
Updated